libpromises/evalfunction.c

```diff
@@ -1124,7 +1124,25 @@ static StringSet *VariablesMatching(const EvalContext *ctx, VariableTableIterato
 
             if (pass)
             {
-                StringSetAdd(matching, expr);
+                JsonElement *data = NULL;
+                bool allocated = false;
+                if (collect_full_data)
+                {
+                    data = VarRefValueToJson(ctx, fp, v->ref, NULL, 0, true, &allocated);
+                }
+
+                if (data == NULL)
+                {
+                    JsonObjectAppendString(matching, expr, "");
```

Comment:
Can you explain this one? When can it happen? And why do we need to append the empty string?